ui/page/components/sub_page.go

```diff
@@ -50,55 +51,87 @@ func (sp *SubPage) Layout(window app.WindowNavigator, gtx layout.Context) layout
+					return sp.Header(window, gtx)
+				})
+			})
+		}),
+		layout.Rigid(func(gtx C) D {
+			return layout.Inset{Top: values.MarginPadding8}.Layout(gtx, sp.Body)
+		}),
+	)
+}
+
+func (sp *SubPage) Header(window app.WindowNavigator, gtx C) D {
 	sp.EventHandler(window)
 
-	return layout.Flex{Axis: layout.Horizontal, Alignment: layout.Middle}.Layout(gtx,
-		layout.Rigid(func(gtx layout.Context) layout.Dimensions {
-			return layout.Inset{
-				Right: values.MarginPadding20,
-			}.Layout(gtx, sp.BackButton.Layout)
-		}),
-		layout.Rigid(func(gtx layout.Context) layout.Dimensions {
-			return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
-				layout.Rigid(sp.Load.Theme.Label(values.TextSize20, sp.Title).Layout),
+	return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+		layout.Rigid(func(gtx C) D {
+			return layout.Flex{Axis: layout.Horizontal, Alignment: layout.Middle}.Layout(gtx,
 				layout.Rigid(func(gtx C) D {
-					if !utils.StringNotEmpty(sp.SubTitle) {
-						return D{}
-					}
-
-					sub := sp.Load.Theme.Label(values.TextSize14, sp.SubTitle)
-					sub.Color = sp.Load.Theme.Color.GrayText2
-					return sub.Layout(gtx)
+					return layout.Inset{
+						Right: values.MarginPadding20,
+					}.Layout(gtx, sp.BackButton.Layout)
 				}),
-			)
-		}),
-		layout.Flexed(1, func(gtx layout.Context) layout.Dimensions {
-			return layout.E.Layout(gtx, func(gtx C) D {
-				if sp.InfoTemplate != "" {
-					return sp.InfoButton.Layout(gtx)
-				} else if sp.ExtraItem != nil {
-					return layout.Flex{Axis: layout.Horizontal}.Layout(gtx,
-						layout.Rigid(func(gtx layout.Context) layout.Dimensions {
-							if sp.ExtraText != "" {
-								return layout.Inset{Right: values.MarginPadding10}.Layout(gtx, func(gtx C) D {
-									return sp.Theme.Caption(sp.ExtraText).Layout(gtx)
-								})
+				layout.Rigid(func(gtx C) D {
+					return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+						layout.Rigid(sp.Load.Theme.LabelSemiBoldSize(values.TextSize20, sp.Title).Layout),
+						layout.Rigid(func(gtx C) D {
+							if !utils.StringNotEmpty(sp.SubTitle) {
+								return D{}
 							}
-							return layout.Dimensions{}
-						}),
-						layout.Rigid(func(gtx layout.Context) layout.Dimensions {
-							return sp.ExtraItem.Layout(gtx, sp.Extra)
+
+							sub := sp.Load.Theme.Label(values.TextSize14, sp.SubTitle)
+							sub.Color = sp.Load.Theme.Color.GrayText2
+							return sub.Layout(gtx)
 						}),
 					)
-				}
-				return layout.Dimensions{}
-			})
+				}),
+				layout.Flexed(1, func(gtx C) D {
+					return layout.E.Layout(gtx, func(gtx C) D {
+						if sp.InfoTemplate != "" {
+							return sp.InfoButton.Layout(gtx)
+						} else if sp.ExtraItem != nil {
+							return layout.Flex{Axis: layout.Horizontal}.Layout(gtx,
+								layout.Rigid(func(gtx C) D {
+									if sp.ExtraText != "" {
+										return layout.Inset{Right: values.MarginPadding10}.Layout(gtx, func(gtx C) D {
+											return sp.Theme.Caption(sp.ExtraText).Layout(gtx)
+										})
```

Comment:
No need for wrapping `sp.Theme.Caption(sp.ExtraText).Layout`.

Reply:
can you explain more?

Reply:
‘return layout.Inset{Right: values.MarginPadding10}.Layout(gtx, sp.Theme.Caption(sp.ExtraText).Layout)’